lib/ProductOpener/Redis.pm

```diff
@@ -74,6 +117,196 @@ sub init_redis() {
 	return;
 }
 
+=head2 subscribe_to_redis_streams ()
+
+Subscribe to redis stream to be informed about user deletions.
+
+=cut
+
+sub subscribe_to_redis_streams () {
+	if (!$redis_url) {
+		# No Redis URL provided, we can't push to Redis
+		if (!$sent_warning_about_missing_redis_url) {
+			$log->warn("Redis URL not provided for streaming") if $log->is_warn();
+			$sent_warning_about_missing_redis_url = 1;
+		}
+		return;
+	}
+
+	if (!defined $redis_client) {
+		# we where deconnected, try again
+		$log->info("Trying to reconnect to Redis") if $log->is_info();
+		init_redis();
+	}
+
+	if (!defined $redis_client) {
+		$log->warn("Can't connect to Redis") if $log->is_warn();
+		return;
+	}
+
+	if (get_oidc_implementation_level() >= 4) {
+		# Read Keycloak events to process actions following user creation / deletion
+		_read_user_streams('$');
+	}
+
+	return;
+}
+
+sub _read_user_streams($search_from) {
+	# Listen for user-deleted events so that we can redact product contributions for this flavor
+	my @streams = ('BLOCK', 0, 'STREAMS', 'user-deleted');
```

Comment:
We should maybe block for a while here and also add a count (same as off-query which uses COUNT 1000 BLOCK 5000)